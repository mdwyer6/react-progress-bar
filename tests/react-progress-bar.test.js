//----------- Jest Tests -----------//

import React from 'react';
import ProgressBar from '../dist/progressbar-bundle.js';
import renderer from 'react-test-renderer';


it('should be able to run tests', () => {
    expect(1 + 2).toEqual(3);
});

it('renders correctly', () => {
  const tree = renderer
    .create(<ProgressBar percent={50} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders when min, max and current are used', () => {
  const tree = renderer
    .create(<ProgressBar min={0} max={20} current={15} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('rounds non integer numbers', () => {
  const tree = renderer
    .create(<ProgressBar percent={51.6573} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('will not render values outside range', () => {
  const tree = renderer
    .create(<ProgressBar percent={155} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('will not render when values are not numbers', () => {
  const tree = renderer
    .create(<ProgressBar percent={'houseboat'} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders themes', () => {
  const tree = renderer
    .create(<ProgressBar percent={51.6573} theme={'arctic'} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('does not render when provided percent and min, max or current values', () => {
  const tree1 = renderer
    .create(<ProgressBar percent={51} min={2} />)
    .toJSON();
  expect(tree1).toMatchSnapshot();

  const tree2 = renderer
    .create(<ProgressBar percent={51} max={2} />)
    .toJSON();
  expect(tree2).toMatchSnapshot();

  const tree3 = renderer
    .create(<ProgressBar percent={51} current={2} />)
    .toJSON();
  expect(tree3).toMatchSnapshot();
});


