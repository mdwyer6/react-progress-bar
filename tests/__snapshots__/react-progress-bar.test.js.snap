// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not render when provided percent and min, max or current values 1`] = `null`;

exports[`does not render when provided percent and min, max or current values 2`] = `null`;

exports[`does not render when provided percent and min, max or current values 3`] = `null`;

exports[`renders correctly 1`] = `
<div
  className="progress-bar-container default"
>
  <div
    className="progress-bar-inner"
    style={
      Object {
        "width": "50%",
      }
    }
  >
    50
    %
  </div>
</div>
`;

exports[`renders themes 1`] = `
<div
  className="progress-bar-container arctic"
>
  <div
    className="progress-bar-inner"
    style={
      Object {
        "width": "52%",
      }
    }
  >
    52
    %
  </div>
</div>
`;

exports[`renders when min, max and current are used 1`] = `
<div
  className="progress-bar-container default"
>
  <div
    className="progress-bar-inner"
    style={
      Object {
        "width": "75%",
      }
    }
  >
    75
    %
  </div>
</div>
`;

exports[`rounds non integer numbers 1`] = `
<div
  className="progress-bar-container default"
>
  <div
    className="progress-bar-inner"
    style={
      Object {
        "width": "52%",
      }
    }
  >
    52
    %
  </div>
</div>
`;

exports[`will not render values outside range 1`] = `null`;

exports[`will not render when values are not numbers 1`] = `null`;
